data/account/participationRegistry.go

```diff
@@ -873,17 +940,23 @@ func (db *participationDB) GetAll() []ParticipationRecord {
 	return results
 }
 
-// GetForRound fetches a record with all secrets for a particular round.
-func (db *participationDB) GetForRound(id ParticipationID, round basics.Round) (ParticipationRecordForRound, error) {
-	var result ParticipationRecordForRound
+func (db *participationDB) GetStateProofForRound(id ParticipationID, round basics.Round) (StateProofRecordForRound, error) {
```

Comment:
Missing doc string

Reply:
It looks like half this function is pretty much the same as `GetForRound`. Could it be simplified by calling that then fetching the missing state proof secret?

Reply:
That's a good idea, `GetForRound` originally contained the logic for both but we've decided that we should only get the StateProof when request instead of every round, since there's some IO overhead and it should only be called once every 256 rounds (as well as caching being problematic as I've explained below).